core/src/test/java/com/google/cloud/sql/core/CloudSqlInstanceTest.java

```diff
@@ -15,52 +15,207 @@
  */
 package com.google.cloud.sql.core;
 
-import static com.google.common.truth.Truth.assertThat;
-import static org.junit.Assert.assertThrows;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
-import com.google.auth.oauth2.GoogleCredentials;
-import com.google.auth.oauth2.OAuth2Credentials;
+import com.google.cloud.sql.AuthType;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.truth.Truth;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import java.io.IOException;
+import java.security.KeyPair;
+import java.sql.Date;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import org.junit.After;
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
 
 @RunWith(JUnit4.class)
 public class CloudSqlInstanceTest {
 
-  @Mock private GoogleCredentials googleCredentials;
-  @Mock private GoogleCredentials scopedCredentials;
-  @Mock private OAuth2Credentials oAuth2Credentials;
+  @Mock private SqlAdminApiFetcher fetcher;
+  @Mock private InstanceData data;
+  @Mock private SslData sslData;
+
+  private int refreshCount;
+  private ListeningScheduledExecutorService executorService;
+  private ListenableFuture<KeyPair> keyPairFuture;
+
+  private StubCredentialFactory stubCredentialFactory =
+      new StubCredentialFactory("my-token", System.currentTimeMillis() + 3600L);
+
+  private CloudSqlInstance instance;
 
   @Before
-  public void setup() {
+  public void setup() throws Exception {
     MockitoAnnotations.openMocks(this);
-    when(googleCredentials.createScoped("https://www.googleapis.com/auth/sqlservice.login"))
-        .thenReturn(scopedCredentials);
+    MockAdminApi mockAdminApi = new MockAdminApi();
+    this.keyPairFuture = Futures.immediateFuture(mockAdminApi.getClientKeyPair());
+    executorService = newTestExecutor();
+    when(data.getSslData()).thenReturn(sslData);
+    refreshCount = 0;
+  }
+
+  @After
+  public void teardown() {
+    executorService.shutdownNow();
+  }
+
+  @Test
+  public void testCloudSqlInstance() throws Exception {
```

Comment:
Could we be more verbose in the test method name? At a glance, I don't know what we're testing here.

Reply:
Fixed.